app/src/main/java/com/imashnake/animite/features/searchbar/SearchFrontDrop.kt

```diff
@@ -0,0 +1,307 @@
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.imashnake.animite.R
+import com.imashnake.animite.core.extensions.landscapeCutoutPadding
+import com.imashnake.animite.core.ui.Icon
+import com.imashnake.animite.core.ui.IconButton
+import com.imashnake.animite.core.ui.TextField
+import com.imashnake.animite.dev.internal.Constants
+import com.imashnake.animite.features.ui.MediaSmall
+import com.imashnake.animite.type.MediaType
+
+/**
+ * Search bar along with a Front Drop list.
+ *
+ * @param hasExtraPadding if the search bar should have extra bottom padding to accommodate the
+ * [com.imashnake.animite.features.navigationbar.NavigationBar].
+ * @param onItemClick called when media with an ID and [MediaType] is clicked.
+ * @param modifier the [Modifier] to be applied to this Front Drop.
+ * @param viewModel [SearchViewModel] instance.
+ */
+@OptIn(ExperimentalAnimationApi::class, ExperimentalLayoutApi::class)
+@Composable
+fun SearchFrontDrop(
+    hasExtraPadding: Boolean,
+    onItemClick: (Int?, MediaType) -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: SearchViewModel = viewModel()
+) {
+    val searchMediaType = MediaType.ANIME
+
+    viewModel.setMediaType(searchMediaType)
+
+    val searchList by viewModel.searchList.collectAsState()
+
+    var isExpanded by rememberSaveable { mutableStateOf(false) }
+    val searchBarBottomPadding: Dp by animateDpAsState(
+        targetValue = if (hasExtraPadding) {
+            dimensionResource(R.dimen.navigation_bar_height)
+        } else 0.dp
+    )
+    val frontDropColor by animateColorAsState(
+        targetValue = MaterialTheme.colorScheme.background.copy(
+            alpha = if (isExpanded) 0.95f else 0f
+        ),
+        animationSpec = tween(Constants.CROSSFADE_DURATION)
+    )
+
+    Box(Modifier.fillMaxSize().drawBehind { drawRect(frontDropColor) })
+
+    searchList.data?.let {
+        SearchList(
+            searchList = it,
+            modifier = Modifier
+                .imeNestedScroll()
+                .landscapeCutoutPadding(),
+            onItemClick = { id ->
+                isExpanded = false
+                viewModel.setQuery(null)
+                onItemClick(id, searchMediaType)
+            }
+        )
+    }
+
+    Surface(
+        color = MaterialTheme.colorScheme.primary,
+        modifier = modifier
+            .landscapeCutoutPadding()
+            .padding(bottom = searchBarBottomPadding)
+            .navigationBarsPadding()
+            .consumeWindowInsets(
+                PaddingValues(bottom = searchBarBottomPadding)
+            )
+            .imePadding()
```

Comment:
Not really, we require that `consumeWindowInsets` is sandwiched between `padding` and `imePadding` since we don't want `padding` when `imePadding` is applied. See [this discussion](https://discord.com/channels/178391822186315777/779143922848628747/1056499628058615930). It removes a lot of sus from #68.

Reply:
We could probably handle it with a custom layout in the future.